### PR TITLE
[Relay] Enhance relay.split(), allow splitted dim to be dynamic

### PR DESCRIPTION
**Pull request by lixiaoquan**

### CODE REVIEW

File: include/tvm/topi/transform.h [hunk 1]
```diff
@@ -481,26 +481,29 @@ inline Tensor stack(const Array<Tensor>& inputs, int axis = 0, std::string name
...
 
-  auto src_axis_size = static_cast<int>(GetConstInt(x->shape[axis]));
-  std::vector<int> begin_ids;
+  auto src_axis_size = x->shape[axis];
+  std::vector<PrimExpr> begin_ids;
   begin_ids.push_back(0);
 
-  for (Integer idx : split_indices) {
-    int val = static_cast<int>(idx->value);
-    CHECK_GT(val, begin_ids.back()) << "split_indices must be sorted";
```
Comment (icemelon9) [inline include/tvm/topi/transform.h:497]: Could you add this check to the shape function?

File: include/tvm/topi/transform.h [hunk 2]
```diff
@@ -668,15 +671,18 @@ inline Array<Tensor> split_sections(const Tensor& x, int num_sections, int axis,
   }
   CHECK_LT(axis, x->shape.size()) << "axis out of bounds";
 
-  auto src_axis_size = static_cast<int>(GetConstInt(x->shape[axis]));
+  auto src_axis_size = x->shape[axis];
 
   CHECK_GT(num_sections, 0) << "Slice count must be > 0";
-  CHECK_EQ(src_axis_size % num_sections, 0)
```
Comment (icemelon9) [inline include/tvm/topi/transform.h:674]: same here

File: include/tvm/topi/transform.h (hunk 1 above)
Comment (lixiaoquan) [reply include/tvm/topi/transform.h:497]: asserted add. The CI error 'test_xgb_model' should not be caused by this change, split() is not used in this test

**Comment by jroesch** [comment]
cc @mbrookhart

**Comment by lixiaoquan** [comment]
@ZihengJiang  Could you help to merge this? Thanks a lot

**Comment by ZihengJiang** [comment]
@lixiaoquan Let's wait for the CI :)

**Comment by lixiaoquan** [comment]
> @lixiaoquan Let's wait for the CI :)

It seems CI didn't report the status, Let me tirgger it again.

**Comment by lixiaoquan** [comment]
@ZihengJiang Could you please take a look? Thanks

**Comment by ZihengJiang** [comment]
@lixiaoquan Merged! Thanks for waiting!

**Pull request merged**